core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -142,6 +142,68 @@ class LogCleanerManagerTest extends JUnitSuite with Logging {
     assertEquals("should have 1 logs ready to be deleted", 1, deletableLog3.size)
   }
 
+  /**
+    * test log race condition between log retention,
+    * topic deletion, and log truncation
+    */
+  @Test
+  def testLogsRaceCondition(): Unit = {
```

Comment:
The test method name `testLogsRaceCondition` is not very informative. Can you follow the existing test names and make it a bit more self-explanatory? It may also be useful to make the comment a bit more informative similar to the comment for `testLogsWithRetentionInprogressShouldNotPickedUpForCompactionAndViceVersa`. The current commit is a bit casual.